starlette/requests.py

```diff
@@ -108,9 +127,10 @@ def user(self) -> typing.Any:
 
     @property
     def state(self) -> State:
-        if "state" not in self._scope:
-            self._scope["state"] = State()
-        return self._scope["state"]
+        if not hasattr(self, '_state'):
+            # Create a state instance with a reference to the dict in which it should store info
```

Comment:
Let's move the `self._scope.setdefault('state', {})` to here.

Reply:
If I move that here, my multi middleware test failed.